### ISSUE TITLE
[1.10.2] [IC2] ic2.core.energy.leg.EnergyNetLocalLeg.tickEnd crash with Energy Storage Module

### ISSUE THREAD

**Issue by tabepic** [micdoodle8/Galacticraft#2546]
Not quite sure why this happened; seems to have something to do with the way power works?

Crash log: https://pastebin.com/hCRZ6Hnv
Forge 12.18.3.2272
Mod list: https://pastebin.com/26VxAmda
Galacticraft Build 61

**Comment by MJRLegends** [comment]
Crash is with the TileEntityEnergyStorageModule

**Comment by tabepic** [comment]
Update: When I went back onto the world, the block affected was missing.

**Comment by tabepic** [comment]
Probably a separate issue, but it seems that glass fibre cable (from IC2) doesn't transmit energy (I was using glass fibre cable in the above crash)

**Comment by radfast** [comment]
You recently updated from an older beta build.
I think this is most likely the issue explained on the forum thread - did you create a new world or are you using your old world?

If you're still using your old world, please just break **all** the Galacticraft machine blocks (including all wires), place them back in the world, and that will probably fix it.

**Comment by tabepic** [comment]
Ok, I'll try that. Thanks

**Comment by radfast** [comment]
hmm, your other issue was from build 58.   If you were already playing build 58 OK then you won't be affected by that save incompatibility issue talked about on the forum.

This problem seems related to IC2, but the real problem is somehow the block has been removed on the server, but IC2 "thinks" it can still find a block there.

I'm not sure how this can happen.

It could be a general problem between IC2 and all of our machines, I haven't tested IC2 in version 1.10.2 at all, yet.

**Comment by tabepic** [comment]
Yeah, i made a new world, and the glass fibre still doesn't work

**Comment by radfast** [comment]
ok, you could try another couple of machines, but I think it's likely IC2 just won't work with Galacticraft yet in 1.10+

**Comment by tabepic** [comment]
Found out how to reproduce: connect glass fibre cable to a Galanticraft machine that uses power and then destroy the machine; it then crashes

**Comment by radfast** [comment]
thanks, perfect - that's very specific, so we should be able to fix this quickly  (but not today, I have to sleep now)

**Comment by tabepic** [comment]
No problem (and good night).

**Comment by radfast** [comment]
I've spent many hours investigating this.  Turns out it's a bug in IC2, during their ```EnergyTileUnloadEvent``` (which happens **after** the block is broken) their code is refreshing the energy network including the broken block, and therefore still asking the broken block if it can receive or emit energy.  They didn't do that in 1.8.9 and before, and I don't know why they changed it.

Anyhow, we have included a workaround.

And it's all good, because in my testing today I have noticed a lot of small ways in which GC was not working properly with IC2, and now it is - everything IC2 should be smooth from build 62.

And Galacticraft restored to its status as (probably) the only mod which can interface IC2 and RF energy. :)
